source/components/defining-a-component.md

```diff
@@ -81,3 +81,57 @@ example, if you have a component called `blog-post`, you would create a
 file at `app/components/blog-post.js`. If your component was called
 `audio-player-controls`, the file name would be at
 `app/components/audio-player-controls.js`.
+
+### Dynamically rendering a component
+
+The `{{component}}` helper can be used to defer the selection of a component to
+run time. The `{{my-component}}` syntax would always render the same component, whereas using the `{{component}}` helper allows swapping the component rendered on the fly.
```

Comment:
Can you provide some real-world cases where you may want to do this? For example:

> “This is useful in cases where, for example, you want to…"